compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala

```diff
@@ -997,6 +1032,11 @@ object PrepJSInterop {
     /** A non-native JS object. */
     val JSMod = new OwnerKind(0x20)
 
+    // Modifiers - these add information to a base kind
+
+    /** A modifer for `JSClass` to signal a non-native JS trait */
+    val WithJSTrait = new OwnerKind(0x1000)
```

Comment:
Instead of introducing this concept of "modifier", I would rather introduce `JSTrait` as new base kind, sitting between `JSClass` and `JSMod`.

Note that `JSClass` and `JSMod` are never directly used for tests, so it only requires adapting the definitions of `JSNonNative` and `AnyClass` below to include `JSTrait` in addition to `JSClass`.

Reply:
Perhaps also rename `JSClass` into `JSNonTraitClass` to make things extra clear (not muddying what "class" means).

Reply:
thanks, much cleaner now :)